### PR TITLE
MS2: On-chain docs

### PR DESCRIPTION
**Pull request by euonymos**
I update the original spec to reflect our current state. I think several conditions might be relaxed, specifically, we can allow some transactions protected by the necessity to put in some ada (like buying a lot) to be performed by anyone as long as all parties get expected outputs. 

I expect the document will evolve slightly in the next milestones, so we don't need to address these questions immediately.

### CODE REVIEW

File: docs/spec.md [hunk 1]
```diff
@@ -0,0 +1,1242 @@
+# Hydra auction blockchain protocol
+
+The Hydra auction blockchain protocol allows
+to run closed auctions on both L1 and L2 Hydra Heads.
+
+It allows delegates to announce themselves on-chain
+offering sellers to use Hydra Heads they manage;
```
Comment (SamDelaney) [inline docs/spec.md]: Commas here, not semicolons.

File: docs/spec.md [hunk 2]
```diff
@@ -0,0 +1,1242 @@
...
+sellers to announce an auction;
+and bidders to enter auctions they are interested in
```
Comment (SamDelaney) [inline docs/spec.md]:
```suggestion
bidders to enter auctions they are interested in
```

File: docs/spec.md [hunk 3]
```diff
@@ -0,0 +1,1242 @@
...
+are both parametrized by the owners's pubkey hash.
+Both sellers and bidders can use oracles.
+
+The auction minting policy is a one-shot policy
+parametrized on the auction metadata validator
+and on the utxo reference of an input (utxo nonce)
+to the auction announcing the transaction.
+The policy atomically mints (and burns)
+three non-fungible tokens.
+One of then, __auction metadata token__,
```
Comment (SamDelaney) [inline docs/spec.md]:
```suggestion
One of them, __auction metadata token__,
```

File: docs/spec.md [hunk 4]
```diff
@@ -0,0 +1,1242 @@
...
+### Auction terms
+
+An auction's behaviour is characterized by its **auction terms**,
+which are fixed when it is announced.
+
+```haskell
+data AuctionTerms = AuctionTerms
+  { at'AuctionLot :: AssetClass
+  -- ^ NFT being sold in the auction.
+  , at'SellerAddress :: Adddress
```
Comment (SamDelaney) [inline docs/spec.md]:
```suggestion
  , at'SellerAddress :: Address
```

File: docs/spec.md [hunk 5]
```diff
@@ -0,0 +1,1242 @@
...
+  , at'BiddingStart :: POSIXTime
+  -- ^ Start time of the bidding period.
+  , at'BiddingEnd :: POSIXTime
+  -- ^ End time of the bidding period.
+  , at'PurchaseDeadline :: POSIXTime
+  -- ^ Time by which the winning bidder can buy the auction lot.
+  -- At and after this time, the winning bidder forfeits its bidder deposit
+  -- if the auction lot has not been purchased.
+  , at'Cleanup :: POSIXTime
+  -- ^ Time at and after  which the remaining utxos in the auction
```
Comment (SamDelaney) [inline docs/spec.md]:
```suggestion
  -- ^ Time at and after which the remaining utxos in the auction
```

File: docs/spec.md [hunk 6]
```diff
@@ -0,0 +1,1242 @@
...
+  -- by this increment.
+  , at'MinDepositAmount :: Integer
+  -- ^ Minimal amount of ADA that the seller requests
+  -- each bidder to place as a bidder deposit for the auction.
+  -- This is only enforced off-chain at the seller's discretion.
+  }
+```
+
+Auction terms are validated on-chain upon minting
+auction metadata and state tokens and cannot be changed.
```
Comment (SamDelaney) [inline docs/spec.md]:
```suggestion
. Auction metadata and state tokens and cannot be changed.
```

File: docs/spec.md [hunk 7]
```diff
@@ -0,0 +1,1242 @@
...
+and place it under his personal oracle validator
+into a utxo containing the list in the datum.
+The seller can do this as many times as needed,
+publishing different lists of signatures each time.
+
+To discover the bidder-auction authorizations from the seller,
+query the seller's personal oracle address
+and filter out utxos that don't contain
+the seller's personal oracle token
+and/or don't relate to the auction based on `auctionId`
```
Comment (SamDelaney) [inline docs/spec.md]:
```suggestion
and/or don't relate to the auction based on `auctionId`.
```

**Review by SamDelaney — APPROVED** [review]
Phenomenal work! I spotted a few typos, but nothing that should delay merging this pr unless you want it to.

**Review by errfrom — APPROVED** [review]

**Pull request merged**